website/addons/wiki/model.py

```diff
@@ -34,6 +35,47 @@ class AddonWikiNodeSettings(AddonNodeSettingsBase):
 
     complete = True
     has_auth = True
+    is_publicly_editable = fields.BooleanField(default=False, index=True)
+
+    def set_editing(self, permissions, auth=None, node=None, save=False, log=False):
+        """Set the editing permissions for this node.
+
+        :param permissions: A string, either 'public' or 'private'
+        :param auth: All the auth informtion including user, API key
+        :param node: Node to which self belongs,
+         only necessary if logging is desired
+        :param bool save: Whether to save the privacy change
+        :param bool log: Whether to add a NodeLog for the privacy change
+        """
+
+        if permissions == 'public' and not self.is_publicly_editable:
+            self.is_publicly_editable = True
+        elif permissions == 'private' and self.is_publicly_editable:
+            self.is_publicly_editable = False
+        else:
+            return False
```

Comment:
These return values are only used in the tests. I'd recommend omitting them, and having the method return nothing.

The tests can check that the state of the object was changed as appropriate instead.